src/ansys/fluent/core/utils/search.py

```diff
@@ -212,10 +238,320 @@ def inner(tree, path, root_path):
-                    type_ = "Object" if isinstance(v, Mapping) else v
                     print(f"{next_path} ({type_})")
             if isinstance(v, Mapping):
                 inner(v, next_path, root_path)
 
     inner(api_tree, "", root_path)
+
+    def _write_api_tree_file(
+        api_objects: list,
+        name: Optional[bool] = None,
+        text: Optional[bool] = False,
+        tui: Optional[bool] = False,
+    ):
+        api_tree_file = get_api_tree_file_name(text=text, name=name, tui=tui)
+        api_tree_file.touch()
+        with open(api_tree_file, "w") as file:
+            for api_object in api_objects:
+                file.write(f"{api_object}")
+                file.write("\n")
+
+    text_file_folder = Path(os.path.join(pyfluent.CODEGEN_OUTDIR, "api_tree"))
+    text_file_folder.mkdir(parents=True, exist_ok=True)
+
+    _write_api_tree_file(api_objects=api_objects, text=True)
+    _write_api_tree_file(api_objects=api_tui_objects, tui=True)
+    _write_api_tree_file(api_objects=api_object_names, name=True)
+
+
+def _get_api_object_names():
+    """Get API object names."""
+    return [
+        line.rstrip("\n")
+        for line in open(get_api_tree_file_name(name=True), "r", encoding="utf-8")
+    ]
+
+
+def _print_search_results(queries: list, wildcard: bool):
+    """Print search results.
+
+    Parameters
+    ----------
+    queries: list
+        List of search string to matche API object names.
+    wildcard: bool
+        Whether to use wildcard search.
+    """
+    api_tree = get_api_tree_file_name(text=True)
+    api_tui_tree = get_api_tree_file_name(tui=True)
+    text_files = [api_tree, api_tui_tree]
+    for text_file in text_files:
+        api_objects = [
+            line.rstrip("\n") for line in open(text_file, "r", encoding="utf-8")
+        ]
+        tfidf_vectorizer = TfidfVectorizer()
+        tfidf_matrix = tfidf_vectorizer.fit_transform(api_objects)
+        for query in queries:
+            query_vector = tfidf_vectorizer.transform([query])
+            cosine_similarities = cosine_similarity(query_vector, tfidf_matrix)
+            if wildcard:
+                most_similar_api_object_index = cosine_similarities.argmax()
+                if query in api_objects[most_similar_api_object_index]:
+                    print(api_objects[most_similar_api_object_index])
+            else:
+                most_similar_api_object_indices = cosine_similarities.argsort()
+                for most_similar_api_object_index in reversed(
+                    most_similar_api_object_indices[0][-3:]
+                ):
+                    if query in api_objects[most_similar_api_object_index]:
+                        print(api_objects[most_similar_api_object_index])
+
+
+def _get_wildcard_matches_for_word_from_names(word: str, names: list):
+    """Get wildcard matches for the given word.
+
+    Parameters
+    ----------
+    word: str
+        The word to search for.
```

Comment:
```suggestion
       Word to search for.
```
We omit leading articles (The, A, An) from brief descriptons so that they start with a significant word.

Reply:
Done. Thank you.